### PR TITLE
fix(security) sanitize host geo coords 

### PR DESCRIPTION
**Pull request by loiclau**
## Description

fix sql injection for host geo coords in form and clapi

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x
- [ ] 18.10.x
- [x] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreonHost.class.php [hunk 1]
```diff
@@ -83,6 +83,7 @@ class CentreonHost extends CentreonObject
     const ORDER_HOSTGROUP = 5;
     const MISSING_INSTANCE = "Instance name is mandatory";
     const UNKNOWN_NOTIFICATION_OPTIONS = "Invalid notifications options";
+    const UNKNOWN_GEO_COORDS = "Invalid geo coords";
```
Comment (sc979) [inline www/class/centreon-clapi/centreonHost.class.php:86]: your constante name does not meet the value
maybe INVALID_GEO_COORDS is more consistent

Comment (sc979) [inline www/class/centreon-clapi/centreonHost.class.php:86]:
```suggestion
    const INVALID_GEO_COORDS = "Invalid geo coords";
```

File: www/class/centreon-clapi/centreonHost.class.php [hunk 2]
```diff
@@ -617,6 +618,9 @@ public function initUpdateParameters($parameters = null)
                     $params[2] = $tpObj->getTimeperiodId($params[2]);
                     break;
                 case "geo_coords":
+                    if (!$this->validateLatLong($params[2])) {
+                        throw new CentreonClapiException(self::UNKNOWN_GEO_COORDS);
```
Comment (sc979) [inline www/class/centreon-clapi/centreonHost.class.php:622]:
```suggestion
                        throw new CentreonClapiException(self::INVALID_GEO_COORDS);
```

File: www/include/common/common-Func.php [hunk 3]
```diff
@@ -2301,3 +2301,19 @@ function get_child($id_page, $lcaTStr)
...
+ */
+function validateGeoCoords()
+{
+    global $form;
+    $coords = $form->getElementValue('geo_coords');
+    return preg_match(
+        '/^[-]?(([0-8]?[0-9])\.(\d+))|(90(\.0+)?),[-]?((((1[0-7][0-9])|([0-9]?[0-9]))\.(\d+))|180(\.0+)?)$/',
+        $coords
+    );
+}
```
Comment (sc979) [inline www/include/common/common-Func.php:2319]:
```suggestion
}

```

Comment (sc979) [inline www/include/common/common-Func.php:2319]:
```suggestion
}

```

File: www/include/configuration/configObject/service/formService.php [hunk 4]
```diff
@@ -869,7 +869,10 @@ function myDecodeService($arg)
     "onkeyup" => "this.blur();this.focus();"
 ));
 $form->addElement('text', 'esi_icon_image_alt', _("Alt icon"), $attrsText);
+
+$form->registerRule('validate_geo_coords', 'function', 'validateGeoCoords');
 $form->addElement('text', 'geo_coords', _("Geo coordinates"), $attrsText);
+$form->addRule('geo_coords', _("geo coords not valid"), 'validate_geo_coords');
```
Comment (kduret) [inline www/include/configuration/configObject/service/formService.php:875]:
```suggestion
$form->addRule('geo_coords', _("geo coords are not valid"), 'validate_geo_coords');
```

File: www/include/configuration/configObject/servicegroup/formServiceGroup.php [hunk 5]
```diff
@@ -108,7 +108,10 @@
 $form->addElement('header', 'information', _("General Information"));
 $form->addElement('text', 'sg_name', _("Name"), $attrsText);
 $form->addElement('text', 'sg_alias', _("Description"), $attrsText);
+
+$form->registerRule('validate_geo_coords', 'function', 'validateGeoCoords');
 $form->addElement('text', 'geo_coords', _("Geo coordinates"), $attrsText);
+$form->addRule('geo_coords', _("geo coords not valid"), 'validate_geo_coords');
```
Comment (kduret) [inline www/include/configuration/configObject/servicegroup/formServiceGroup.php:114]:
```suggestion
$form->addRule('geo_coords', _("geo coords are not valid"), 'validate_geo_coords');
```

File: www/include/configuration/configObject/meta_service/formMetaService.php [hunk 6]
```diff
@@ -251,7 +251,9 @@
 $form->setDefaults(array('meta_activate' => '1'));
 $form->addElement('textarea', 'meta_comment', _("Comments"), $attrsTextarea);
 
+$form->registerRule('validate_geo_coords', 'function', 'validateGeoCoords');
 $form->addElement('text', 'geo_coords', _("Geo coordinates"), $attrsText);
+$form->addRule('geo_coords', _("geo coords not valid"), 'validate_geo_coords');
```
Comment (kduret) [inline www/include/configuration/configObject/meta_service/formMetaService.php:256]:
```suggestion
$form->addRule('geo_coords', _("geo coords are not valid"), 'validate_geo_coords');
```

File: www/include/configuration/configObject/hostgroup/formHostGroup.php [hunk 7]
```diff
@@ -157,7 +157,11 @@
  * Further informations
  */
 $form->addElement('text', 'hg_rrd_retention', _('RRD retention'), array('size' => 5));
+
+$form->registerRule('validate_geo_coords', 'function', 'validateGeoCoords');
 $form->addElement('text', 'geo_coords', _("Geo coordinates"), $attrsText);
+$form->addRule('geo_coords', _("geo coords not valid"), 'validate_geo_coords');
```
Comment (kduret) [inline www/include/configuration/configObject/hostgroup/formHostGroup.php:163]:
```suggestion
$form->addRule('geo_coords', _("geo coords are not valid"), 'validate_geo_coords');
```

File: www/include/configuration/configObject/host/formHost.php [hunk 8]
```diff
@@ -965,6 +972,7 @@ function allInSameInstance($hosts, $instanceId)
 $form->addElement('text', 'ehi_2d_coords', _("2d Coords"), $attrsText2);
 $form->addElement('text', 'ehi_3d_coords', _("3d Coords"), $attrsText2);
 $form->addElement('text', 'geo_coords', _("Geo coordinates"), $attrsText);
+$form->addRule('geo_coords', _("geo coords not valid"), 'validate_geo_coords');
```
Comment (kduret) [inline www/include/configuration/configObject/host/formHost.php:975]:
```suggestion
$form->addRule('geo_coords', _("geo coords are not valid"), 'validate_geo_coords');
```

File: www/class/centreon-clapi/centreonHost.class.php [hunk 9]
```diff
@@ -83,6 +83,7 @@ class CentreonHost extends CentreonObject
     const ORDER_HOSTGROUP = 5;
     const MISSING_INSTANCE = "Instance name is mandatory";
     const UNKNOWN_NOTIFICATION_OPTIONS = "Invalid notifications options";
+    const INVALID_GEO_COORDS = "Invalid geo coords";
```
Comment (kduret) [inline www/class/centreon-clapi/centreonHost.class.php:86]:
```suggestion
    public const INVALID_GEO_COORDS = "Invalid geo coords";
```

File: www/class/centreon-clapi/centreonHostGroup.class.php [hunk 10]
```diff
@@ -56,6 +56,7 @@ class CentreonHostGroup extends CentreonObject
 {
     const ORDER_UNIQUENAME = 0;
     const ORDER_ALIAS = 1;
+    const INVALID_GEO_COORDS = "Invalid geo coords";
```
Comment (kduret) [inline www/class/centreon-clapi/centreonHostGroup.class.php:59]:
```suggestion
    public const INVALID_GEO_COORDS = "Invalid geo coords";
```

File: www/class/centreon-clapi/centreonService.class.php [hunk 11]
```diff
@@ -76,6 +76,7 @@ class CentreonService extends CentreonObject
     const ORDER_SVCTPL = 2;
     const NB_UPDATE_PARAMS = 4;
     const UNKNOWN_NOTIFICATION_OPTIONS = "Invalid notifications options";
+    const INVALID_GEO_COORDS = "Invalid geo coords";
```
Comment (kduret) [inline www/class/centreon-clapi/centreonService.class.php:79]:
```suggestion
    public const INVALID_GEO_COORDS = "Invalid geo coords";
```

File: www/class/centreon-clapi/centreonServiceGroup.class.php [hunk 12]
```diff
@@ -52,6 +52,7 @@ class CentreonServiceGroup extends CentreonObject
 {
     const ORDER_UNIQUENAME = 0;
     const ORDER_ALIAS = 1;
+    const INVALID_GEO_COORDS = "Invalid geo coords";
```
Comment (kduret) [inline www/class/centreon-clapi/centreonServiceGroup.class.php:55]:
```suggestion
    public const INVALID_GEO_COORDS = "Invalid geo coords";
```

File: www/class/centreon-clapi/centreonUtils.class.php [hunk 13]
```diff
@@ -114,6 +114,17 @@ public static function convertLineBreak($str)
         return $str;
     }
 
+    /**
+     * @param $coords
```
Comment (callapa) [inline www/class/centreon-clapi/centreonUtils.class.php:118]: Can you give the correct format for $coords?

File: www/class/centreon-clapi/centreonUtils.class.php [hunk 14]
```diff
@@ -114,6 +114,17 @@ public static function convertLineBreak($str)
...
+     * @return false|int
```
Comment (callapa) [inline www/class/centreon-clapi/centreonUtils.class.php:119]: If you just validate, just return a boolean value, not int.

File: www/include/common/common-Func.php [hunk 15]
```diff
@@ -2301,3 +2301,19 @@ function get_child($id_page, $lcaTStr)
...
     return $redirect;
 }
+
+/**
+ * Quickform rule that validate geo_coords
+ *
+ * @return false|int
+ * @throws HTML_QuickForm_Error
+ */
+function validateGeoCoords()
```
Comment (callapa) [inline www/include/common/common-Func.php:2311]: Why create the same method twice?

Comment (loiclau) [reply www/include/common/common-Func.php:2311]: one is specific to quickform

**Pull request merged**